recipes/cpr/all/conanfile.py

```diff
@@ -91,19 +96,37 @@ def _patch_sources(self):
         for patch in self.conan_data.get("patches", {}).get(self.version, []):
             tools.patch(**patch)
 
+    def _get_cmake_option(self, option):
+        CPR_1_6_CMAKE_OPTIONS_TO_OLD = {
+            "CPR_FORCE_USE_SYSTEM_CURL": "USE_SYSTEM_CURL",
+            "CPR_BUILD_TESTS": "BUILD_CPR_TESTS",
+            "CPR_BUILD_TESTS_SSL": "BUILD_CPR_TESTS_SSL",
+            "CPR_GENERATE_COVERAGE": "GENERATE_COVERAGE",
+            "CPR_USE_SYSTEM_GTEST": "USE_SYSTEM_GTEST",
+            "CPR_FORCE_OPENSSL_BACKEND": "USE_OPENSSL",
+            "CPR_FORCE_WINSSL_BACKEND": "USE_WINSSL",
+            "CPR_GENERATE_COVERAGE": "GENERATE_COVERAGE",
+        }
+
+        if self._uses_old_cmake_options:
+            # Get the translated option if we can, or the original if one isn't defined.
+            return CPR_1_6_CMAKE_OPTIONS_TO_OLD.get(option, option)
+        else:
+            return option
+
     def _configure_cmake(self):
         if not self._cmake:
             self._cmake = CMake(self)
-            self._cmake.definitions["USE_SYSTEM_CURL"] = True
-            self._cmake.definitions["BUILD_CPR_TESTS"] = False
-            self._cmake.definitions["GENERATE_COVERAGE"] = False
-            self._cmake.definitions["USE_SYSTEM_GTEST"] = False
+            self._cmake.definitions[self._get_cmake_option("CPR_FORCE_USE_SYSTEM_CURL")] = True
+            self._cmake.definitions[self._get_cmake_option("CPR_BUILD_TESTS")] = False
+            self._cmake.definitions[self._get_cmake_option("CPR_GENERATE_COVERAGE")] = False
+            self._cmake.definitions[self._get_cmake_option("CPR_USE_SYSTEM_GTEST")] = False
             if self._supports_openssl:
                 self._cmake.definitions["CMAKE_USE_OPENSSL"] = self.options.get_safe("with_openssl", False)
             if self._supports_winssl: # The CMake options changed
                 # https://github.com/whoshuu/cpr/commit/18e1fc5c3fc0ffc07695f1d78897fb69e7474ea9#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR39-R40
-                self._cmake.definitions["USE_OPENSSL"] = self.options.get_safe("with_openssl", False)
-                self._cmake.definitions["USE_WINSSL"] = self.options.get_safe("with_winssl", False)
+                self._cmake.definitions[self._get_cmake_option("CPR_FORCE_OPENSSL_BACKEND")] = self.options.get_safe("with_openssl", False)
+                self._cmake.definitions[self._get_cmake_option("CPR_FORCE_WINSSL_BACKEND")] = self.options.get_safe("with_winssl", False)
```

Comment:
The logs complain about `CPR_ENABLE_SSL`.
It looks like that value needs to be set as well.

I think OPENSSL/WINSSL is only used on windows to choose between winssl and openssl.

Reply:
What's interesting is that this only fails on MacOS. Changing `CPR_ENABLE_SSL` would disable SSL support, which doesn't seem like a good idea. I don't have a Mac available to me at the moment, though. I need to think of some way to dig into this...

Reply:
Isn't disabling ssl possible if both `with_openssl` ànd `with_winsll` are False?

Reply:
I'm not saying it's not possible, as much as it might not be wise to do so.

Reply:
It looks like this is the result of an upstream bug in CPR. Indeed, I can replicate this exact error on a Mac. Going to wait to see how they resolve this before I patch in my own solution.

https://github.com/whoshuu/cpr/issues/546

Reply:
You can go ahead here by throwing an `InvalidConfiguration` when both `ssl` options are set to False and add a link to the issue you mentioned above.